frontend/src/modules/EditZing/Components/EmailModal.tsx

```diff
@@ -54,29 +51,12 @@ export const EmailModal = ({
   )
 
   useEffect(() => {
-    axios
-      .get(`${API_ROOT}${EMAIL_PATH}/templates`)
-      .then(async (res: AxiosResponse<EmailTemplatesResponse>) => {
-        const allTemplates = await Promise.all(
-          res.data.data
-            .map(responseEmailTemplateToEmailTemplate)
-            .map(async (template: EmailTemplate, i) => {
-              // Download the HTML for the email body in Cloud Storage bucket
-              const url = await getDownloadURL(
-                ref(templatesBucket, template.body)
-              )
-              const html = (await axios.get(url)).data as string
-              return { ...template, html }
-            })
-        )
-        const filteredTemplates = allTemplates.filter(
-          (template) => template.type === recipientType
-        )
-        setTemplates(filteredTemplates)
-        setSelectedTemplate(filteredTemplates[0])
-      })
-      .catch((error) => console.error(error))
-  }, [recipientType])
+    const filteredTemplates = templates.filter(
+      (template) => template.type === recipientType
+    )
+    setFilteredTemplates(filteredTemplates)
+    setSelectedTemplate(filteredTemplates[0])
+  }, [recipientType, templates])
```

Comment:
I don't think these need to be in a useEffect hook anymore because email templates will always be loaded if this modal is able to appear, so `templates` will be available on component initialization.